src/data/codingcontracttypes.ts

```diff
@@ -1297,19 +1297,21 @@ export const codingContractTypesMetadata: ICodingContractTypeMetadata[] = [
     difficulty: 5,
     desc: (n: unknown): string => {
       return [
-        "You are given the following decimal Value: \n",
-        `${n} \n`,
-        "Convert it to a binary representation and encode it as an 'extended Hamming code'. Eg:\n ",
-        "Value 8 is expressed in binary as '1000', which will be encoded",
-        "with the pattern 'pppdpddd', where p is a parity bit and d a data bit. The encoding of\n",
-        "8 is 11110000. As another example, '10101' (Value 21) will result into (pppdpdddpd) '1001101011'.\n",
-        "The answer should be given as a string containing only 1s and 0s.\n",
-        "NOTE: the endianness of the data bits is reversed in relation to the endianness of the parity bits.\n",
-        "NOTE: The bit at index zero is the overall parity bit, this should be set last.\n",
-        "NOTE 2: You should watch the Hamming Code video from 3Blue1Brown, which explains the 'rule' of encoding,",
-        "including the first index parity bit mentioned in the previous note.\n\n",
-        "Extra rule for encoding:\n",
-        "There should be no leading zeros in the 'data bit' section",
+        "You are given the following decimal value: \n",
+        `${n} \n\n`,
+        "Convert it to a binary representation and encode it as an 'extended Hamming code'.\n ",
+        "The number should be converted to a string of '0' and '1' with no leading zeroes.\n",
+        "Parity bits are inserted at positions 0 and 2^N.\n",
+        "Parity bits are used to make the total number of '1' bits in a given set of data even.\n",
+        "The parity bit at position 0 considers all bits including parity bits.\n",
+        "Each parity bit at position 2^N alternately considers N bits then ignores N bits, starting at position 2^N.\n",
+        "The endianness of the parity bits is reversed compared to the endianness of the data bits:\n",
+        "Data bits are encoded left-to-right (most significant byte first) and the parity bits encoded right-to-left.\n",
+        "The parity bit at position 0 is set last.\n\n",
+        "Examples:\n",
+        "8 in binary is 1000, and encodes to 11110000 (pppdpddd - where p is a parity bit and d is a data bit)\n",
+        "21 in binary is 10101, and encodes to 1001101011 (pppdpdddpd)\n\n",
+        "For more information on the 'rule' of encoding, refer to the Wikipedia entry and the 3Blue1Brown videos on Hamming Codes.",
```

Comment:
```suggestion
        "For more information on the 'rule' of encoding, refer to Wikipedia (https://wikipedia.org/wiki/Hamming_code)",
        "or the 3Blue1Brown videos on Hamming Codes. (https://youtube.com/watch?v=X8jsijhllIA)",
```
Adding the links explicitly to make it easy for people (same for the decoding side)